### PR TITLE
feat: adds draft status

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Adds "draft" status to Research Updates, matched against design pattern for Howto form. A separate PR will be add the ability to delete Research Updates.

⚠️ Marking an update as Draft makes it unreachable, as Draft items are not listed on the Research Article page. One possible workaround for this is being discussed here: #2218 

## Git Issues

Addresses #2199 

## Screenshots/Videos

Screenshot 2023-04-05 at 21 23 10

Screenshot 2023-04-05 at 22 40 38

### CODE REVIEW

**Comment by davehakkens** [comment]
Seems like its missing the part to go back and continue on the draft?
After saving as a draft I don't how to find it back, edit an publish?
Or am I not seeing it @thisislawatts?

File: src/pages/Research/Content/ResearchArticle.test.tsx [hunk 1]
```diff
@@ -102,6 +102,46 @@ describe('Research Article', () => {
...
+        </ThemeProvider>
+      </Provider>,
+    )
+
+    // Assert
+    expect(() => {
+      wrapper.getByText('Research Update #2')
+    }).toThrow()
+  })
 })
```
Comment (AlfonsoGhislieri) [inline src/pages/Research/Content/ResearchArticle.test.tsx:145]: **Suggestion**: Should we maybe include an assertion to check that 'Research Update #1' is showing just to be a little more thorough

File: src/pages/Research/Content/Common/ResearchUpdate.form.tsx [hunk 2]
```diff
@@ -321,14 +303,35 @@ const UpdateForm = observer((props: IProps) => {
...
                   }}
                 >
+                  <Button
+                    data-cy={'draft'}
+                    onClick={(evt) => {
+                      trySubmitForm(true)
+                      evt.preventDefault()
+                    }}
+                    variant="secondary"
+                    type="submit"
```
Comment (AlfonsoGhislieri) [inline src/pages/Research/Content/Common/ResearchUpdate.form.tsx:313]: **Question**: Why is `evt.preventDefault()` needed now compared to before?

Comment (thisislawatts) [reply src/pages/Research/Content/Common/ResearchUpdate.form.tsx:313]: Good catch, thanks! Removed.

File: src/pages/Research/Content/ResearchArticle.test.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Research/Content/ResearchArticle.test.tsx:146]: Added ✨

File: packages/components/src/ResearchEditorOverview/ResearchEditorOverview.stories.tsx [hunk 3]
```diff
@@ -17,7 +17,7 @@ Default.args = {
       isActive: true,
       title: 'Update 1',
       slug: 'update-1',
-      status: 'draft',
+      status: 'published',
     },
     {
```
Comment (AlfonsoGhislieri) [inline packages/components/src/ResearchEditorOverview/ResearchEditorOverview.stories.tsx:22]: **Suggestion**: Could be good to have a story that shows some updates with the draft status

File: src/pages/Research/Content/Common/ResearchUpdate.form.tsx [hunk 4]
```diff
@@ -376,6 +367,29 @@ export const ResearchUpdateForm = observer((props: IProps) => {
...
+    isCreating
+      ? {
+          isActive: false,
+          title: researchTitle,
+          status: 'draft',
+          slug: null,
+        }
+      : null,
+  ].filter(Boolean)
+
```
Comment (AlfonsoGhislieri) [inline src/pages/Research/Content/Common/ResearchUpdate.form.tsx:392]: **Praise**: nice refactoring here, good to see this added use of a type and pulling all this logic out of the main component return!

**Review by AlfonsoGhislieri — APPROVED** [review]

File: src/pages/Research/Content/Common/ResearchUpdate.form.tsx [hunk 5]
```diff
@@ -343,18 +346,12 @@ const UpdateForm = observer((props: IProps) => {
...
-                        })),
-                        {
-                          isActive: false,
-                          title: values.title || 'Title of this update',
-                          slug: null,
-                        },
-                      ]}
+                      updates={getResearchUpdates(
+                        store.activeResearchItem.updates,
+                        props.formValues._id,
```
Comment (AlfonsoGhislieri) [inline src/pages/Research/Content/Common/ResearchUpdate.form.tsx:351]: **Issue**: Should we not be checking or guarding `store.activeResearchItem.updates ` being undefined here, could it not cause an error if we try and map over it after?

**Review by AlfonsoGhislieri — CHANGES_REQUESTED** [review]

File: src/pages/Research/Content/Common/ResearchUpdate.form.tsx (hunk 5 above)
Comment (thisislawatts) [reply src/pages/Research/Content/Common/ResearchUpdate.form.tsx:351]: Added https://github.com/ONEARMY/community-platform/pull/2206/commits/87137b47cd97e309749c1acf2202173ce6a43a63

**Review by AlfonsoGhislieri — APPROVED** [review]
Looks good to me @thisislawatts thanks for adding the fallback, just a conflict to resolve!

**Pull request merged**